Add support of nSIM for testing Picolibc

kolerov opened the pull request:
Support of nSIM for testing Picolibc is already in Picolibc repository. These changes must be added to scripts to use nSIM with Picolibc testsuite:

1. Generate a correct cross file with nSIM wrapper for running binaries (used only for testing).
2. Use a different memory layout for nSIM - put everything below `0x8000000` to make it possible to build all tests both for `rv32` and `rv64` targets. Note that QEMU's memory for code starts from `0x80000000`.
3. Pass `-Dtest-machine=nsim` to Picolibc configurator to trigger nSIM specific configurations.

Also, a few small patches included.

kolerov committed:
picolibc: Add support of nSIM to cross file generator

Picolibc for ARC-V uses run-riscv-nsim script for running
tests on nSIM. It must be specified as a wrapper in
a cross file when nSIM is chosen as a simulator for
testing.

Signed-off-by: Yuriy Kolerov <ykolerov@synopsys.com>

kolerov committed:
picolibc: Set default section size to 128K as in Newlib

Signed-off-by: Yuriy Kolerov <ykolerov@synopsys.com>

kolerov committed:
picolibc: Use different memory layouts for release and testing

1. Release build must use memory layout from Newlib.
2. Memory layout for testing must depend on a chosen
   simulator (nSIM or QEMU). Also, section size must
   be increased.

Signed-off-by: Yuriy Kolerov <ykolerov@synopsys.com>

kolerov committed:
picolibc: Pass -Dtest-machine={qemu,nsim} to test builds

Picolibc may use different simulators for testing
and QEMU is default. The latest Picolibc for ARC-V
contains support of nSIM.

Signed-off-by: Yuriy Kolerov <ykolerov@synopsys.com>

kolerov committed:
picolibc: Do not fail when "ninja test" fails

Ninja test fails when at least a single test fails. However,
non-0 return code prevents creating a stamp file.

Signed-off-by: Yuriy Kolerov <ykolerov@synopsys.com>

kolerov committed:
arcv: Update heads for submodules

Signed-off-by: Yuriy Kolerov <ykolerov@synopsys.com>

artemiy-volkov commented on Makefile.in:
```diff
@@ -1157,7 +1191,7 @@ stamps/build-picolibc-nano-test: stamps/configure-picolibc-nano-test
 
 stamps/check-libc-picolibc-nano: stamps/build-picolibc-nano-test $(SIM_STAMP)
-	ninja -C build-picolibc-nano-test test
+	ninja -C build-picolibc-nano-test test || true
```
looks a bit hacky, why the "|| true"?

kolerov replied:
If `ninja test` fails then `make` fails too and it will not save a stamp file. `ninja test` return non 0 exit code (a number of failed tests) if any tests fails. You can find this hack in other places:

https://github.com/foss-for-synopsys-dwc-arc-processors/arcv-gnu-toolchain/blob/arc-2025.06/Makefile.in#L1703

artemiy-volkov replied:
Fair enough, thank you for the explanation.

kolerov committed:
picolibc: Use meson test instead of ninja test

This allows to avoid extra messages in output when
there are failing tests:

    FAILED: meson-internal__test
    /u/ykolerov/.local/bin/meson test --no-rebuild --print-errorlogs
    ninja: build stopped: subcommand failed.

Signed-off-by: Yuriy Kolerov <ykolerov@synopsys.com>

kolerov commented:
Added a small update.
